SmartAppointmentSystem.Business/Implementations/AppointmentService.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Microsoft.EntityFrameworkCore;
+using SmartAppointmentSystem.Business.Contracts;
+using SmartAppointmentSystem.Business.DTOs;
+using SmartAppointmentSystem.Business.Extensions;
+using SmartAppointmentSystem.Data;
+using SmartAppointmentSystem.Data.Entities;
+
+namespace SmartAppointmentSystem.Business.Implementations;
+
+public class AppointmentService(AppointmentContext context) : IAppointmentService
+{
+    public async Task CreateAppointmentAsync(AppointmentRequestDTO appointmentRequestDTO, CancellationToken cancellationToken)
```

Comment:
It could be better to have a default value for cancellation token on methods `CancellationToken cancellationToken = default`. If you do not pass a default value, you can't call this method without a cancellationToken. Both works fine, it's up to you!